app/models/topic.rb

```diff
@@ -1106,6 +1113,8 @@ def make_banner!(user)
     self.save
 
     MessageBus.publish('/site/banner', banner)
+
+    Jobs.enqueue_at(bannered_until, :remove_banner, topic_id: self.id) if bannered_until
```

Comment:
Should we at least check that `bannered_until` is in the future?

Reply:
We need to add an ensure consistency job like how we did for `pinned_until`. Sidekiq can be wiped and jobs can fail so we need some way to recovering here.

Reply:
Yes I know about the risk but didn't know we had this for pinned, cool :+1: will add it

I will need to add a bannered_until field on topics table

Reply:
For the pinning functionality we cancel the existing job first before scheduling a new one, I think we should do that here too:

```ruby
Jobs.cancel_scheduled_job(:remove_banner, topic_id: self.id)
```